server/models/listing.model.js

```diff
@@ -20,17 +19,12 @@ async function newListing(listing){
         })
     }
         try {
-            //Pseudo code for userId (role=shelter) check
 
-//Before attempting to create a listing via Prisma we have to perform a check on the userId provided.
-// If the provided userId has a role=shelter then they are allowed to create a listing
-// Otherwise we return an error : User does not have access rights for this.
-
-// So i would add another await method to find user by id 
-//(probably something like findUnique) and the check the returned record that indeed it has the shelter role
-// If the returned role is correct the we proceed to the create
-            const isUserRoleShelter = await prisma.user.findUnique(listing.userId)
-            console.log(isUserRoleShelter)
+            const isUserRoleShelter = await prisma.user.findUnique({
+                where: {
+                    id: listing.userId,
+                  },
+            })
             if(isUserRoleShelter.role === "shelter") {
                 const newListing = await  prisma.listing.create({
                     data: {
```

Comment:
Why are you passing, to prisma.listing.create, the id: listing.id ? Isn't this id automatically generated  by the DB?

Reply:
Hi,

My Suggestion:

Resource Id is part of data contract for all API post requests.

Is optional (on the WebAPI data contract, not on the DB). 
If it is in the request's body, will be validated - must be a valid uuid, unique, etc - and used. If not, the backend will generate a valid uuid.
The UI client (web, mobile, etc) do not fill this if field (will be generated on the BE).

But, and we can discuss that, when creating test data sets (directs calls to API), or moving data to other DBs, maybe it can be useful to hardcode the Id's.
Examples:
(1) seed data for Prisma 
(2) a test Postman collection - posting the listing, we want to use a well-known UserId (previously hardecoded), I think 😉

That´s just an old habit, we can debate 🙂